lib/bson/bson.js

```diff
@@ -1282,8 +1284,13 @@ BSON.deserialize = function(buffer, options, isArray) {
           if(subType == Binary.SUBTYPE_BYTE_ARRAY) {
             binarySize = buffer[index++] | buffer[index++] << 8 | buffer[index++] << 16 | buffer[index++] << 24;
           }
-          // Slice the data
-          object[name] = new Binary(buffer.slice(index, index + binarySize), subType);
+          if(nativeBuffers) {
+            // Slice the data and create a native Buffer object
+            object[name] = new Buffer(buffer.slice(index, index + binarySize));
```

Comment:
`buffer.slice()` returns a buffer, why not just use that? Now you are doing a memory copy. https://github.com/nodejs/io.js/blob/master/lib/buffer.js#L96-L100